internal/cephfs/util.go

```diff
@@ -97,3 +124,50 @@ func (cs *ControllerServer) validateExpandVolumeRequest(req *csi.ControllerExpan
 
 	return nil
 }
+
+// cephfsSnapshot represents a CSI snapshot and its cluster information.
+type cephfsSnapshot struct {
+	NamePrefix   string
+	Monitors     string
+	MetadataPool string
+	Pool         string
+	ClusterID    string
+	RequestName  string
+}
+
+func genSnapFromOptions(ctx context.Context, volOpt *volumeOptions, snapOptions map[string]string) *cephfsSnapshot {
+	var err error
+
+	cephfsSnap := &cephfsSnapshot{}
+	cephfsSnap.Pool = volOpt.Pool
+	cephfsSnap.MetadataPool = volOpt.MetadataPool
+
+	cephfsSnap.Monitors, cephfsSnap.ClusterID, err = getMonsAndClusterID(ctx, snapOptions)
+	if err != nil {
+		cephfsSnap.Monitors = volOpt.Monitors
+		cephfsSnap.ClusterID = volOpt.ClusterID
+	}
+
+	if namePrefix, ok := snapOptions["snapshotNamePrefix"]; ok {
+		cephfsSnap.NamePrefix = namePrefix
+	}
+
+	return cephfsSnap
+}
+
+func getMonsAndClusterID(ctx context.Context, options map[string]string) (monitors, clusterID string, err error) {
```

Comment:
the function already exists in RBD we need to move it to utils and use  it

Reply:
marking this in todo or tracker.. Also will wait for #1224